Foundation/NSString.swift

```diff
@@ -1214,7 +1214,17 @@ extension NSString {
     }
     
     public convenience init(format: String, locale: AnyObject?, arguments argList: CVaListPointer) {
-        NSUnimplemented()    
+        let str: CFString
+        if let loc = locale {
+            if loc.dynamicType === NSLocale.self || loc.dynamicType === NSDictionary.self {
+                str = CFStringCreateWithFormatAndArguments(kCFAllocatorSystemDefault, unsafeBitCast(loc, to: CFDictionary.self), format._cfObject, argList)
+            } else {
+                fatalError("locale parameter must be a NSLocale or a NSDictionary")
+            }
+        } else {
+            str = CFStringCreateWithFormatAndArguments(kCFAllocatorSystemDefault, nil, format._cfObject, argList)
+        }
+        self.init(str._swiftObject)
```

Comment:
The one possible issue I could see with this implementation is that it wont call the description with locale on the NS objects: _CFStringCreateWithFormatAndArgumentsAux2 can pass a function pointer to allow for fetching the descriptionWithLocale() off of the objects. However that would mean that we would need to have an internal protocol to identify those classes who respond to that method.